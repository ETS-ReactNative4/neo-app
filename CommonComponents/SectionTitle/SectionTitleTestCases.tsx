import React from "react";
import { ITestCase } from "../../TypeInterfaces/TestCases/ITestCases";
import SectionTitle from "./SectionTitle";

const SectionTitleTestCases: ITestCase[] = [
  {
    title: "Section Title",
    Component: <SectionTitle title={"Live on-trip support"} />
  },
  {
    title: "Section Title With Short Description",
    Component: (
      <SectionTitle
        title={"Live on-trip support"}
        description={
          "We’ll use your preference info to make better and more relevant recommendations."
        }
      />
    )
  },
  {
    title: "Section Title With Longer Description",
    Component: (
      <SectionTitle
        title={"Live on-trip support"}
        description={
          "We’ll use your preference info to make better and more relevant recommendations. We’ll use your preference info to make better and more relevant recommendations."
        }
      />
    )
  }
];

export default SectionTitleTestCases;
