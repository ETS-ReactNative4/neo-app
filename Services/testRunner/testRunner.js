import renderer from "react-test-renderer";

const testRunner = testCase => {
  test(testCase.title, () => {
    const tree = renderer.create(testCase.Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export default testRunner;
