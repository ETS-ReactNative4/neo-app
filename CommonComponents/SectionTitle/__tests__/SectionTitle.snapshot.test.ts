import SectionTitleTestCases from "../SectionTitleTestCases";
import testRunner from "../../../Services/testRunner/testRunner";

SectionTitleTestCases.map(testRunner);
