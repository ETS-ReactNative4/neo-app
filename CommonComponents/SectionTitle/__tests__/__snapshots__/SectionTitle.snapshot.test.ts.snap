// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Title 1`] = `
<View>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "rgba(44, 47, 66, 1)",
          "fontFamily": "SourceSansPro-SemiBold",
          "fontSize": 20,
          "lineHeight": 24,
          "marginBottom": 8,
        },
        undefined,
      ]
    }
  >
    Live on-trip support
  </Text>
</View>
`;

exports[`Section Title With Longer Description 1`] = `
<View>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "rgba(44, 47, 66, 1)",
          "fontFamily": "SourceSansPro-SemiBold",
          "fontSize": 20,
          "lineHeight": 24,
          "marginBottom": 8,
        },
        undefined,
      ]
    }
  >
    Live on-trip support
  </Text>
  <Text
    ellipsizeMode="tail"
    numberOfLines={2}
    style={
      Array [
        Object {
          "color": "rgba(83, 87, 109, 1)",
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 14,
          "lineHeight": 20,
        },
        undefined,
      ]
    }
  >
    We’ll use your preference info to make better and more relevant recommendations. We’ll use your preference info to make better and more relevant recommendations.
  </Text>
</View>
`;

exports[`Section Title With Short Description 1`] = `
<View>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "rgba(44, 47, 66, 1)",
          "fontFamily": "SourceSansPro-SemiBold",
          "fontSize": 20,
          "lineHeight": 24,
          "marginBottom": 8,
        },
        undefined,
      ]
    }
  >
    Live on-trip support
  </Text>
  <Text
    ellipsizeMode="tail"
    numberOfLines={2}
    style={
      Array [
        Object {
          "color": "rgba(83, 87, 109, 1)",
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 14,
          "lineHeight": 20,
        },
        undefined,
      ]
    }
  >
    We’ll use your preference info to make better and more relevant recommendations.
  </Text>
</View>
`;

exports[`Section Title with custom text styles 1`] = `
<View>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "rgba(44, 47, 66, 1)",
          "fontFamily": "SourceSansPro-SemiBold",
          "fontSize": 20,
          "lineHeight": 24,
          "marginBottom": 8,
        },
        Object {
          "color": "black",
          "fontFamily": "SourceSansPro-SemiBold",
          "fontSize": 30,
          "lineHeight": 36,
        },
      ]
    }
  >
    Live on-trip support
  </Text>
  <Text
    ellipsizeMode="tail"
    numberOfLines={2}
    style={
      Array [
        Object {
          "color": "rgba(83, 87, 109, 1)",
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 14,
          "lineHeight": 20,
        },
        Object {
          "color": "black",
        },
      ]
    }
  >
    We’ll use your preference info to make better and more relevant recommendations. We’ll use your preference info to make better and more relevant recommendations.
  </Text>
</View>
`;
